### PR TITLE
Update math_util.c and math_util.h to support autodoc with trig functions (inline instead of macros) and add new functions to smlua_math_utils

### PR DESCRIPTION
**Pull request by Multi-Volt**
Hello,

This pull request continues the work from my previous PR, which can be found [here](https://github.com/coop-deluxe/sm64coopdx/pull/345#event-14580671499). The original repository was unintentionally messed up due to complications with `format.sh` and me being new to git, necessitating this re-implementation. In this update, I have replaced several macros in `math_util.h` with inline functions, which provide a slight [runtime efficiency improvement](https://en.wikipedia.org/wiki/Inline_expansion). Inline functions also offer enhanced type safety compared to macros, and I have introduced additional macros to support multiple data types, minimizing the need for any kind of overloading.

This PR also introduces new functions in `smlua_math_utils.c`, aimed at assisting modders with various calculations. The existing, unused functions in `misc.c` relating to angle conversions have been added here so modders can leverage the Lua math library's trigonometric functions when needed. Based on feedback from a recent poll on the Discord server, I added numerical conversions to support working in degrees as well, which was the majority's preferred format. Additionally, I implemented a basic function for computing the hypotenuse of two points, incorporating a variant of the fast square root algorithm (from Quake). These features are similar to ones already present in [SRB2's Lua implementation](https://wiki.srb2.org/wiki/Lua/Functions).

Poll:
image

Example of performance increase (extreme case):

Before inlining:
Screenshot from 2024-10-09 14-48-31

After inlining:
Screenshot from 2024-10-09 14-48-00

Please let me know if any further adjustments or formatting changes are needed.

### CODE REVIEW

**Review by Isaac0-dev — COMMENTED** [review]
The performance increase looks fairly significant. 
Most of the changes to `misc.c` are formatting changes, and some make the code more difficult to read than before imo.

**Comment by Multi-Volt** [comment]
> The performance increase looks fairly significant. Most of the changes to `misc.c` are formatting changes, and some make the code more difficult to read than before imo.

Yeah that's mostly just a remnant from `format.sh`, I went ahead and reverted the formatting minus the removal of the functions now present in `smlua_math_utils`

**Comment by Multi-Volt** [comment]
> > The performance increase looks fairly significant. Most of the changes to `misc.c` are formatting changes, and some make the code more difficult to read than before imo.
> 
> Yeah that's mostly just a remnant from `format.sh`, I went ahead and reverted the formatting minus the removal of the functions now present in `smlua_math_utils`.

Update: I have went ahead and taken the liberty to revert all changes made by `format.sh`, This should make the changes easier to review.

**Comment by Multi-Volt** [comment]
Update 2: Added even more optimized versions of each function for even more performance gains by utilizing the builtins from GCC/Clang when they are available.

File: src/pc/lua/utils/smlua_math_utils.c [hunk 1]
```diff
@@ -1,28 +1,31 @@
...
 
-s32 max(s32 a, s32 b) {
-    return ((a) > (b) ? (a) : (b));
+f32 sm64_to_degrees(s16 sm64_angle) {
+    return sm64_angle * 180.0f / 0x8000;
 }
 
-f32 maxf(f32 a, f32 b) {
-    return ((a) > (b) ? (a) : (b));
+s16 degrees_to_sm64(f32 degrees_angle) {
```
Comment (Isaac0-dev) [inline src/pc/lua/utils/smlua_math_utils.c]: please use camelCase for the variable names here

**Review by Isaac0-dev — CHANGES_REQUESTED** [review]
sorry I missed this before

File: src/pc/lua/utils/smlua_math_utils.c (hunk 1 above)
Comment (Multi-Volt) [reply src/pc/lua/utils/smlua_math_utils.c]: went ahead and fixed the variable names adding camel case.

**Pull request merged**